src/collections/news/2023-04-15-Baby-Bliss-Bot-AI-Techniques.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Baby Bliss Bot - Experiments with AI
+slug: clusive-latest-features
+date: "2023-04-15"
+excerpt: Exploring AI Machine Learning models with Bliss symbols.
+---
+A number of different artificial intelligence machine learning systems were
+explored in order to acquire a basic knowledge of the systems, how to train
+them, and what they produce, as well as an idea of their value to the
+[Baby Bliss Bot Project](https://github.com/fluid-project/floeproject.org/blob/eb04bf9b0774f452145376f3495933eb3ce81c10/src/collections/news/2023-04-12-BBB-Intro.md).
+ __TODO: Fix link after referenced blog post is merged.__
+
+## StyleGAN: StyleGAN2-ADA and StyleGAN3
+
+A GAN (Generative Adversarial Network) is an AI that generates fake content
+after being trained with a large set of real data.  An example is shown on the
+[Random Face Generator (This Person Does Not Exist)](https://this-person-does-not-exist.com/en),
+web site where the images of faces that are produced are not of actual people.
+Instead the GAN, which has been trained on many pictures of actual faces,
+composes a new face based on what it has learned from its training.
+
+Nvidia has developed a type of a progressive GAN called a [StyleGAN](https://en.wikipedia.org/wiki/StyleGAN).
+The system is trained first with low resolution images, and as training
+proceeds, higher resolutions are progressively introduced.
+
+There are several versions of StyleGAN.  We investigated two of them:
+StyleGAN2-ADA and StyleGAN3.
+
+### StyleGAN2-ADA
+
+The relevant aspect of StyleGAN2-ADA is that this version of StyleGAN can be
+trained using a small data set of data.  In that regard, there are only 1,217
+distinct single Bliss symbols out of 6,000+ total symbols.  Greyscale images
+of the single Bliss symbols were used as the training set.  These are shown
+in the following array of symbols:
+
+<img src="/news/images/StyleGAN2-ADA-reals.png" alt="array of real single
+character Bliss symbols">
+
+Guided by the article [How to Train StyleGAN2-ADA with Custom Dataset](https://towardsdatascience.com/how-to-train-stylegan2-ada-with-custom-dataset-dc268ff70544),
+the 1,216 single Bliss symbols were used to train the network.
+Training was conducted over several runs for total training time of 62 hours.
+The following two figures show arrays of the original real images used for
+training and the final fake images of Bliss symbols generated by the GAN:
+
+<img src="/news/images/StyleGAN2-ADA-fakes.png" alt="array of fake Bliss
+symbol images as generated by StyleGAN2-ADA">
+
+As can be seen, the fake images have a superficial resemblance to real Bliss
+symbols and it appears that the GAN has detected and learned some patterns
+within the single Bliss symbols. At the present time it is unclear what the
+patterns are nor exactly what the GAN has learned.  These are issues for
```

Comment:
remove "nor"?

Reply:
The "nor" sentence structure is awkward.  I will reword.

Reply:
I was wondering if should use the language "fake images" rather than simply "generated images", and adding a note about how these images do not correspond to "real" Bliss symbols but generated images that attempt to extract certain features found in the training data. I just don't seem to like the use of "fake" as it has been so much used and over used in expressions l"fake news" and others... just a thought

Reply:
Point taken @agamba.  I, too, find that the use of "fake" can be pejorative.  However, the GAN technical literature regularly uses the word "fake" and the software itself outputs files literally named "fakes000880.png".  It's clear that their sense of "fake" is the neutral "not real".

Still, casual readers of the blog might interpret "fake" along the lines of "fake news", and there is at least one case where faces generated by StyleGAN have been used to set up fake Face Book accounts.

How about using the word "synthesized"?